Automattic/wp-calypso
# chore: Migrate Desktop to Yarn v3

## Background

While `./desktop` lives in this repo, it is not part of the main dependency tree (i.e. it has its own `yarn.lock` and it is not a yarn workspace). This adds some unnecessary complexity and eventually, we'd like `desktop` to be part of the main dependency tree (i.e. be just another yarn workspace).

## Changes proposed in this Pull Request

To achieve that goal we need to merge the dependency trees, and the first step is to make sure both use the same Yarn version. This PR migrates `./desktop` to use Yarn v3.

## Testing instructions

The most important check, in this case, is that the desktop's dependency tree hasn't changed.

1)  Compute the effective tree for this branch by running:
    ```
    cd desktop
    rm -fr **/node_modules
    yarn
    ../packages/effective-module-tree/cli.js -o list > effective-tree
    ```
2) Repeat the same steps for `trunk`
3) Diff both `effective-tree` files. There should be no differences.
